### PR TITLE
YLP-950 Generate robot.txt and sitemap.xml

### PR DESCRIPTION
**Pull request by yvon-dblg**

### CODE REVIEW

File: server/gen-utils.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  console.log(`Loading ${resolvedTemplateFilename}...`);
+  const templateContent = await fs.readFile(resolvedTemplateFilename, { encoding: "utf-8", flag: "r" });
+
+  console.log("Using configuration", templateValues);
+  const templateCompiler = handlebars.compile(templateContent);
+
+  const outputFileContent = templateCompiler(templateValues);
+  const resolvedOutputFilename = `${getDistDir()}/${outputFilename}`;
+  await fs.writeFile(resolvedOutputFilename, outputFileContent, { encoding: "utf-8", flag: "w", mode: 0o644 });
+  console.log(`Generating ${resolvedOutputFilename} OK`);
```
Comment (franck-fourel) [inline server/gen-utils.js:48]:
```suggestion
  console.log(`${resolvedOutputFilename} generated successfully`);
```

File: server/package.json [hunk 2]
```diff
@@ -7,12 +7,14 @@
...
+    "gen-robot": "node gen-robot-txt.js",
+    "gen-sitemap": "node gen-sitemap-xml.js",
     "gen-lambda": "node cloudfront-gen-lambda.js",
     "security-checks": "npm audit --production --audit-level=high --registry=https://registry.npmjs.org/"
   },
   "engines": {
-    "node": ">=10.x",
-    "npm": ">=6.x"
+    "node": ">=12.x",
+    "npm": ">=7.x"
```
Comment (franck-fourel) [inline server/package.json:17]: is it necessary to make this upgrade here ?

**Review by franck-fourel — COMMENTED** [review]
couple of comments to review together
Wondering as well if we could/should put more links in the sitemap

File: packages/yourloops/components/footer-links.tsx [hunk 3]
```diff
@@ -105,7 +105,7 @@ function FooterLinks(props: FooterLinksProps): JSX.Element {
     <Container id="footer-links-container" className={containerClassName} maxWidth="sm">
       <Grid id="footer-links" container>
         <Grid item xs={4} className={classes.rightLink}>
-          <Link id="footer-link-url-privacy-policy" href={diabeloopUrls.getPrivacyPolicyUrL(i18n.language)}>{t("footer-link-url-privacy-policy")}</Link>
+          <Link id="footer-link-url-privacy-policy" href={diabeloopUrls.getPrivacyPolicyUrL(i18n.language)} rel="nofollow">{t("footer-link-url-privacy-policy")}</Link>
```
[the remark this answers is not in the document]
Comment (yvon-dblg) [reply packages/yourloops/components/footer-links.tsx:108]: Because Marine says so

File: server/package.json (hunk 2 above)
Comment (yvon-dblg) [reply server/package.json:17]: Not really, I can create a specific PR for this ?

File: packages/yourloops/components/footer-links.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (franck-fourel) [reply packages/yourloops/components/footer-links.tsx:108]: sounds like a good reason ;-)

File: server/package.json (hunk 2 above)
Comment (franck-fourel) [reply server/package.json:17]: yep please, that could make more sense to isolate it

**Review by franck-fourel — APPROVED** [review]
LGTM

**Pull request merged**